### PR TITLE
mysqlcluster: when storage size shrink, and change replicas, It will …

### LINKED ISSUE #436: [bug] when storage size shrink, and change replicas, It will reconcile fail forever.
**Describe the problem**

**To Reproduce**

1. set replicas to 2 and storage size to  50Gi
2. then set replica to 3 and storage size to 20Gi
3. it will still at replica 2 , and cluster is nerver ready
image

image

**Expected behavior**

shrink storage size is not allowed, but the reconcile should be successful.

**Environment:**

- RadonDB MySQL version:

### PR DESCRIPTION
**Pull request by acekingke**
…reconcile fail forever. #436

### What type of PR is this?

/bug
### Which issue(s) this PR fixes?

Fixes #436 

### What this PR does?

Summary:

### CODE REVIEW

**Comment by runkecheng** [comment]
Whether the pvc expansion needs to set a maximum number of retry?

File: sidecar/config.go [hunk 1]
```diff
@@ -333,7 +333,7 @@ func (cfg *Config) buildXenonConf() []byte {
 			"admin": "root",
 			"ping-timeout": %d,
 			"passwd": "%s",
-			"host": "localhost",
+			"host": "127.0.0.1",
```
Comment (runkecheng) [inline sidecar/config.go:336]: other pr. fix #440

**Review by runkecheng — APPROVED** [review]

File: mysqlcluster/syncer/statefulset.go [hunk 2]
```diff
@@ -167,8 +171,22 @@ func (s *StatefulSetSyncer) canExpandPVC(ctx context.Context) bool {
 	// If newStorage is not greater than oldStorage, do not expand.
 	if newStorage.Cmp(*oldRequest.Storage()) != 1 {
 		log.Info("canExpandPVC", "result", "can not expand", "reason", "new pvc is not larger than old pvc")
-		return false
+		// Should return true, because if change replicas and storage size, it will fail again and again
+		// during expanding PVCs, not great then oldStorage it will not truely expand, just recreate
```
Comment (andyli029) [inline mysqlcluster/syncer/statefulset.go:175]: great then
->
greater than

**Comment by acekingke** [comment]
use the webhook, forbidden the `shrink size`

**Comment by acekingke** [comment]
use the webhook, forbidden the `shrink size`

**Comment by runkecheng** [comment]
fixed by #487

**Pull request closed without merging**